mlflow/store/artifact/http_artifact_repo.py

```diff
@@ -119,3 +136,47 @@ def abort_multipart_upload(self, local_file, upload_id, artifact_path=None):
         }
         resp = http_request(host_creds, endpoint, "POST", json=params)
         augmented_raise_for_status(resp)
+
+    def _multipart_log_artifact(self, local_file, artifact_path=None) -> bool:
+        """
+        Attempts to perform multipart upload to log an artifact.
+        Returns True if the multipart upload is successful.
+        """
+        parts = []
+        chunk_size = MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.get()
+        size = os.path.getsize(local_file)
+        num_parts = math.ceil(size / chunk_size)
+
+        try:
+            create = self.create_multipart_upload(local_file, num_parts, artifact_path)
+        except HTTPError as e:
+            # return False if server does not support multipart upload
+            error_message = e.response.json().get("message", "")
+            if isinstance(error_message, str) and error_message.startswith(
+                "Multipart upload is not supported for artifact repository"
+            ):
+                return False
+            else:
+                raise
```

Comment:
We can add an error representing MPU not supported if that can make code more readable:

```python
try:
    try_mpu()
except MpuUnsupported:
    pass


# single upload


```

Reply:
good point, updated!